### PR TITLE
feat(sbom): extract component hashes from CycloneDX and SPDX SBOMs

### PR DESCRIPTION
**Pull request by ruromero**
## Summary

- Add `componentHashes` field to `DependencyTree` as `Map<String, Map<String, String>>` (purl → algorithm → hash value)
- Extract all available checksums from CycloneDX components and SPDX packages
- Add tests and SBOM fixtures for both formats

## Jira

[TC-4334](https://redhat.atlassian.net/browse/TC-4334)

## Test plan

- [x] CycloneDX: SHA-256 and MD5 hashes extracted from components
- [x] CycloneDX: components without hashes produce empty map
- [x] SPDX: SHA256 and MD5 checksums extracted from packages
- [x] SPDX: packages without checksums produce empty map
- [x] All 16 tests pass (`mvn test`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[TC-4334]: https://redhat.atlassian.net/browse/TC-4334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by Sourcery

Add support for capturing per-component checksum hashes from CycloneDX and SPDX SBOMs in the dependency tree model.

New Features:
- Expose a componentHashes map on DependencyTree to store checksum hashes per component purl.
- Extract checksum hashes from CycloneDX component entries into the dependency tree.
- Extract checksum hashes from SPDX package entries into the dependency tree via the SpdxWrapper and SpdxParser.

Tests:
- Add CycloneDX parser tests and fixtures to verify hash extraction and behavior when no hashes are present.
- Add SPDX parser tests and fixtures to verify checksum extraction and behavior when no checksums are present.

### CODE REVIEW

File: src/main/java/io/github/guacsec/trustifyda/model/DependencyTree.java [hunk 1]
```diff
@@ -32,7 +32,8 @@
 public record DependencyTree(
     Map<PackageRef, DirectDependency> dependencies,
     Map<String, String> licenseExpressions,
-    PackageRef root) {
+    PackageRef root,
+    Map<String, Map<String, String>> componentHashes) {
 
   public static final PackageRef getDefaultRoot(String type) {
     return PackageRef.builder()
```
[the remark this answers is not in the document]
Comment (ruromero) [reply src/main/java/io/github/guacsec/trustifyda/model/DependencyTree.java:39]: [sdlc-workflow/verify-pr] Classified as **suggestion** — no convention match found; the existing codebase pattern wraps only outer maps with `Collections.unmodifiableMap()`, leaving inner values unwrapped. No sub-task created.

File: src/test/java/io/github/guacsec/trustifyda/integration/backend/sbom/cyclonedx/CycloneDxParserTest.java [hunk 2]
```diff
@@ -141,6 +142,55 @@ void testSupportedVersions(String version) throws IOException {
     }
   }
 
+  /** Verifies that all hashes are extracted from CycloneDX component entries. */
+  @Test
+  void testParseHashesFromComponents() throws IOException {
```
[the remark this answers is not in the document]
Comment (ruromero) [reply src/test/java/io/github/guacsec/trustifyda/integration/backend/sbom/cyclonedx/CycloneDxParserTest.java:147]: [sdlc-workflow/verify-pr] Classified as **suggestion** — edge case test coverage is a quality improvement, not a convention requirement. No sub-task created.

**Comment by ruromero** [comment]
## Verification Report — TC-4334

**Task:** Extract component hashes from CycloneDX and SPDX SBOMs in backend parser
**PR:** #606
**Branch:** TC-4334

### Verdicts

| Check | Verdict | Notes |
|---|---|---|
| Scope Containment | WARN | `CycloneDxParserTest.java` modified but not in task's Files to Modify — benign companion test change |
| Diff Size | PASS | 8 files changed, well within scope |
| Commit Traceability | PASS | All commits trace to TC-4334 |
| Sensitive Patterns | PASS | No secrets, credentials, or sensitive data |
| CI Status | PASS | All checks green |
| Acceptance Criteria | PASS | All 5 acceptance criteria satisfied |
| Verification Commands | PASS | `mvn test` and `mvn spotless:check` pass |
| Test Quality | PASS | Tests are non-repetitive, well-documented with Javadoc |
| Test Change Classification | ADDITIVE | New test file + new test methods only |

### Review Comments

| Comment | Author | Classification | Action |
|---|---|---|---|
| [#3205329424](https://github.com/guacsec/trustify-dependency-analytics/pull/606#discussion_r3205329424) | sourcery-ai | Suggestion | No action — existing codebase wraps outer maps only |
| [#3205329425](https://github.com/guacsec/trustify-dependency-analytics/pull/606#discussion_r3205329425) | sourcery-ai | Suggestion | No action — edge case tests are optional quality improvement |

### Summary

Implementation correctly extracts **all hash algorithms** (not just SHA-256) from both CycloneDX and SPDX SBOMs, storing them as `Map<String, Map<String, String>>` (purl → algorithm → hash value). Algorithm key formats intentionally differ between formats (CycloneDX: `SHA-256`, SPDX: `SHA256`) to reflect each spec's native naming. Test coverage includes both positive (multi-algorithm extraction) and negative (no-hashes) cases for both SBOM formats.

No sub-tasks created. No blocking issues found.

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.8.2.*
